### PR TITLE
Initial commit with react and most popup functionality

### PR DESCRIPTION
**Pull request by jenniferharmon**
PR for Issue #491 

Hi! I stumbled across this issue while browsing for Hacktoberfest.

This Pull Request copies React libs from `node_modules/` to the `js/vendor/` directory. It currently uses the minified production version of React, but can be switched to use the development version.

`popup.js` has been added to .gitignore because it is now auto-generated from `popup.jsx` using babel.

I moved the `ui.js` logic into `popup.jsx`. I noticed vad.js calls ui.setState(). I would be happy to update that to use browser.runtime.sendMessage() and create a new type for changing the popup state.

Do you have any suggestions for how I might improve the Pull Request?

### CODE REVIEW

File: extension/popup/popup.css [hunk 1]
```diff
@@ -15,91 +15,13 @@ body,
   font-family: "Inter" !important;
 }
 
-.listening #voice-input {
```
Comment (ianb) [inline extension/popup/popup.css]: Removing CSS is great! I haven't been doing a great job of that myself :-/

File: extension/popup/popup.jsx [hunk 2]
```diff
@@ -0,0 +1,744 @@
+/* globals util, voice, vad, lottie, settings, log, voiceShim, buildSettings */
+
+const React = window.React
```
Comment (ianb) [inline extension/popup/popup.jsx]: If you put `React` and `ReactDOM` in the `/* globals */` section then these assignments won't be necessary.

File: extension/popup/popup.jsx [hunk 3]
```diff
@@ -0,0 +1,744 @@
...
+        }
+        if (
+            !backgroundTabRecorder &&
+            isWaitingForPermission &&
+            Date.now() - isWaitingForPermission < FAST_PERMISSION_CLOSE
+        ) {
+            this.startOnboarding()
+        }
+
+        // TODO: offload mic and other resources before closing?
```
Comment (ianb) [inline extension/popup/popup.jsx]: Does this all get run even when the window is closed? I don't think so, and so I think most of this cleanup should happen in a `beforeunload` handler. (Probably this component will never unmount?)

File: extension/popup/popup.jsx [hunk 4]
```diff
@@ -0,0 +1,744 @@
...
+    updateExamples = async () => {
+        const suggestions = await browser.runtime.sendMessage({
+            type: "getExamples",
+            number: 3,
+        })
+
+        this.setState({ suggestions })
+    }
+
+    setCurrentState = (currentState) => {
```
Comment (ianb) [inline extension/popup/popup.jsx]: I'm not sure this is the right pattern to be using with React, or at least not in its entirety. I think it makes sense for a couple cases:

1. Waiting for voice
2. Waiting for typing (once typing has begun)
3. After a command is heard/entered.

The third case has a bunch of cases, like a card, an error message, some text, etc. But those can each be properties, and each displayed if the data is present.

That said, it's probably best to leave this as it is for now without trying to change everything on the first pass. This is already a big patch!

**Comment by jenniferharmon** [comment]
Thanks for the great feedback @ianb 

I will try to address these issues in the next day or two.

**Comment by jenniferharmon** [comment]
I believe my latest commit makes it safe to resolve the merge conflicts by accepting my changes to those files. I ported the logic in b5a81bacf6f142ae414a493b5e917540 to popup.jsx. 

I was not able to trigger a scenario where a search card gets updated. Any suggestions for testing that?

**Comment by ianb** [comment]
If the changes are applied, you should be able to merge master and then `git add` the deleted files to resolve the conflict.

For a card, search for "weather".

**Comment by jenniferharmon** [comment]
Hi @ianb I think I have addressed all of your suggestions, with the exception of the unload event handling, which I believe was also not working before the refactor. Please let me know if there is anything else you would like done or changed to accept the PR.

**Comment by ianb** [comment]
Thanks, this is great! There might be some details to deal with, but I think it's easier to do that in-tree given the size of the patch. So we'll get it landed and take it from there.

I can tell you've figured out quite a bit about the project on your own here, I hope we'll be seeing more contributions from you!

**Comment by jenniferharmon** [comment]
Thanks! This was my first time working on a browser extension. I had fun and learned a lot about the project while working on the refactor. Feel free to `@mention` me on any issues you think I might be well-suited to work on.

**Pull request merged**